src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_pending_resumption_handler.cc

```diff
@@ -0,0 +1,253 @@
+#include "rc_rpc_plugin/rc_pending_resumption_handler.h"
+#include "rc_rpc_plugin/rc_helpers.h"
+#include "rc_rpc_plugin/rc_module_constants.h"
+
+namespace rc_rpc_plugin {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "RCPendingResumptionHandler")
+
+RCPendingResumptionHandler::RCPendingResumptionHandler(
+    application_manager::ApplicationManager& application_manager)
+    : ExtensionPendingResumptionHandler(application_manager)
+    , rpc_service_(application_manager.GetRPCService()) {}
+
+void RCPendingResumptionHandler::on_event(
+    const application_manager::event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const auto cid = event.smart_object_correlation_id();
+  LOG4CXX_TRACE(logger_,
+                "Received event with function id: "
+                    << event.id() << " and correlation id: " << cid);
+  sync_primitives::AutoLock lock(pending_resumption_lock_);
+  auto data_request = pending_requests_.front();
+  if (cid != data_request.first) {
```

Comment:
Suppose that cid may be somewhere in the middle of pending_requests_, cant it?

Reply:
@LuxoftAKutsan 
Fixed in:https://github.com/LuxoftSDL/sdl_core/pull/99/commits/114a2b4f4858dd189f26c13f00aa4292489b374b